queue.go

```diff
@@ -0,0 +1,80 @@
+package cleisthenes
+
+import "fmt"
+
+// Define transaction FIFO queue
+type queue struct {
+	txs []Transaction
+}
+
+// EmptyQueErr is for calling peek(), poll() when queue is empty.
+type emptyQueErr struct{}
+
+func (e *emptyQueErr) Error() string {
+	return "Queue is empty."
+}
+
+type indexBoundaryErr struct {
+	queSize int
+	want    int
+}
+
+func (e *indexBoundaryErr) Error() string {
+	return fmt.Sprintf("Index is larger than queue size.\n"+
+		"queue size : %d, you want : %d", e.queSize, e.want)
+}
+
+func NewQueue() queue {
+	return queue{
+		txs: []Transaction{},
+	}
+}
+
+// Check whether queue is empty or not.
+func (q *queue) IsEmpty() bool {
+	return len(q.txs) == 0
+}
+
+// Return first element of queue, but not erase it.
+func (q *queue) Peek() (Transaction, error) {
+	if q.IsEmpty() {
+		return nil, &emptyQueErr{}
+	}
+	return q.txs[0], nil
+}
+
+// Return first element of queue, and erase it.
+func (q *queue) Poll() (Transaction, error) {
+	ret, err := q.Peek()
+	if err != nil {
+		return nil, err
+	}
+	q.txs = q.txs[1:]
+	return ret, nil
+}
+
+// Return size of queue
+func (q *queue) Length() int {
```

Comment:
to `Len()` plz

ref: https://godoc.org/github.com/golang-collections/go-datastructures/queue